src/ddx.h

```diff
@@ -0,0 +1,99 @@
+#ifndef DDX_H
```

Comment:
Is this a generated file?

Reply:
Partly. As written in the comments some of the code is generated by the python script.